### PR TITLE
refactor(Configuration):changed inputs to be FileInput

### PR DESCRIPTION
**Pull request by eyalkenig**
Changed the input format, to allow more input types.
Currently supporting only FileInput type

### CODE REVIEW

**Comment by RonBarabash** [comment]
@eyalkenig im wondering if we should move the inputs to be define inside the metric file, where u define your outputs also..

File: src/test/scala/com/yotpo/metorikku/test/metorikku-test-config-invalid-query.yaml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 metrics:
   - src/test/calculations/invalid_metric_query_test/
 inputs:
- employees: src/test/tables/employees.parquet
+ employees:
+  type: file
```
Comment (lyogev) [inline src/test/scala/com/yotpo/metorikku/test/metorikku-test-config-invalid-query.yaml]: Any way to make this the default behaviour? so you can still remain with:
`employees: src/test...`

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.0.6-SNAPSHOT"
+version in ThisBuild := "1.0.0"
```
Comment (lyogev) [inline version.sbt]: We're not ready to be 1.0.0 yet :)

File: build.sbt [hunk 3]
```diff
@@ -40,7 +40,8 @@ libraryDependencies ++= Seq(
   "com.fasterxml.jackson.core" % "jackson-annotations" % jacksonVersion,
   "com.fasterxml.jackson.core" % "jackson-databind" % jacksonVersion,
   "com.fasterxml.jackson.dataformat" % "jackson-dataformat-yaml" % jacksonVersion,
-  "com.groupon.dse" % "spark-metrics" % "2.0.0"
+  "com.groupon.dse" % "spark-metrics" % "2.0.0",
+  "mysql" % "mysql-connector-java" % "5.1.42"
```
Comment (lyogev) [inline build.sbt]: Please remove this.
We don't want this library bundled with Metorikku, you can write in the Readme that in order to use the JDBC reader the user must add to the spark submit command the following (mysql for example):
--packages mysql:mysql-connector-java:5.1.42

File: src/main/scala/com/yotpo/metorikku/configuration/inputs/JdbcInput.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import com.yotpo.metorikku.input.InputTableReader
+import com.yotpo.metorikku.session.Session.getSparkSession
+import org.apache.spark.sql.DataFrame
+
+case class JdbcInput(var name: String, var connection: Connection, table: String) extends Input {
+  override def getSequence: Seq[String] = Seq(table)
+  override def getReader(seq: Seq[String]): InputTableReader = sqlTableReader
+  private object sqlTableReader extends InputTableReader {
+    def read(tablePaths: Seq[String]): DataFrame = {
+      val url = s"jdbc:mysql://${connection.dbAddress}:${connection.dbPort}/${connection.dbName}?zeroDateTimeBehavior=convertToNull"
```
Comment (lyogev) [inline src/main/scala/com/yotpo/metorikku/configuration/inputs/JdbcInput.scala]: please remove the hardcoded mysql driver from the url

File: src/main/scala/com/yotpo/metorikku/configuration/inputs/JdbcInput.scala [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package com.yotpo.metorikku.configuration.inputs
+
+import com.yotpo.metorikku.configuration.Connection
+import com.yotpo.metorikku.input.InputTableReader
+import com.yotpo.metorikku.session.Session.getSparkSession
+import org.apache.spark.sql.DataFrame
+
+case class JdbcInput(var name: String, var connection: Connection, table: String) extends Input {
```
Comment (RonBarabash) [inline src/main/scala/com/yotpo/metorikku/configuration/inputs/JdbcInput.scala]: why vars and not vals ?

File: version.sbt (hunk 2 above)
Comment (eyalkenig) [reply version.sbt]: if we're introducing breaking changes (like inputs format) - we should bump major version..

**Comment by lyogev** [comment]
I'm claiming it's not a breaking change. Check out my other comments.

**Comment by lyogev** [comment]
Also version bumps are done as part of the release and not commit based.

**Pull request closed without merging**